src/helpers/request-extract.js

```diff
@@ -0,0 +1,38 @@
+import Promise from "bluebird";
+import URI from "urijs";
+
+/**
+ * Start an extract job and be notified with the url when complete.
+ * @param {Object} ctx The Context Object
+ * @param  {Object} options
+ * @return {Promise}
+ */
+export default function requestExtract(ctx, { segment = null, format = "json", path = "batch", fields = [] }) {
+  const { client, hostname } = ctx;
+  const search = client.configuration();
+  if (segment) {
+    search.segment_id = segment.id;
+  }
+  const url = URI(`https://${hostname}`)
```

Comment:
I would pass the hostname explicitly here. Not use the ctx directly.
`hostname` is bot available in a worker context, right ?